src/main/java/com/ebivariation/contigalias/service/AssemblyService.java

```diff
@@ -16,36 +16,123 @@
 
 package com.ebivariation.contigalias.service;
 
-import com.ebivariation.contigalias.dao.AssemblyDao;
+import com.ebivariation.contigalias.datasource.AssemblyDataSource;
 import com.ebivariation.contigalias.entities.AssemblyEntity;
 import com.ebivariation.contigalias.entities.ChromosomeEntity;
+import com.ebivariation.contigalias.repo.AssemblyRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
 import java.io.IOException;
+import java.util.LinkedList;
 import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 @Service
 public class AssemblyService {
 
-    private final AssemblyDao assemblyDao;
+    private static final IllegalArgumentException duplicateAssemblyInsertionException = new IllegalArgumentException(
```

Comment:
this is not common, but I would be fine with it in other cases. In this case, the exception would be clearer if you include the accession that is already present in the DB (or even the whole AssemblyEntity).